pkg/providers/amifamily/ami.go

```diff
@@ -117,6 +119,9 @@ func NewDefaultProvider(versionProvider version.Provider, ssm ssmiface.SSMAPI, e
 
 // Get Returning a list of AMIs with its associated requirements
 func (p *DefaultProvider) Get(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, options *Options) (AMIs, error) {
+	p.Lock()
+	defer p.Unlock()
```

Comment:
Do we need to write lock? what are we writing here? It should just be a read lock on L139, right? Can we just do it then?

Reply:
We need a write lock here because we are writing to the cache, If we don't we could have a cache miss and call AWS more times then needed

Reply:
Yeah let's keep the lock in here